cmd/util/ledger/util/payload_grouping_test.go

```diff
@@ -0,0 +1,135 @@
+package util_test
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	rand2 "math/rand"
+	"runtime"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/util"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func TestGroupPayloadsByAccount(t *testing.T) {
+	log := zerolog.New(zerolog.NewTestWriter(t))
+	payloads := generateRandomPayloads(1000000)
+	tmp := make([]*ledger.Payload, len(payloads))
+	copy(tmp, payloads)
+
+	groups := util.GroupPayloadsByAccount(log, payloads, 0)
+
+	require.Greater(t, groups.Len(), 1)
+}
+
+func TestGroupPayloadsByAccountCompareResults(t *testing.T) {
+	log := zerolog.Nop()
+	payloads := generateRandomPayloads(1000000)
+	tmp1 := make([]*ledger.Payload, len(payloads))
+	tmp2 := make([]*ledger.Payload, len(payloads))
+	copy(tmp1, payloads)
+	copy(tmp2, payloads)
+
+	groups1 := util.GroupPayloadsByAccount(log, tmp1, 0)
+	groups2 := util.GroupPayloadsByAccount(log, tmp2, runtime.NumCPU())
```

Comment:
This test case compare the result with itself, it validates the concurrency behavior, but didn't validate the correctness of the result.

In order to validate the correctness of the result, I think we should also compare with a result generated by using a map without any concurrency.

Reply:
That is what the `groups1` are.
`groups2` are with concurrency  `groups1` are without.

Reply:
Yeah, group2 tests the validation. 

Both group1 and group2 are from the same implementation which uses sort. I'm suggesting to compare with an implementation without sort, such as a simple map.

Reply:
added.